### PR TITLE
Support uploading multiple files together with use_manifest=True.

### PR DESCRIPTION
**Pull request by pkaleta**
@jtratner cc @counsyl/stor 

Before this PR, there was no ability to specify multiple files for upload together with `use_manifest=True`. This PR attempts to fix that.

I've added a unit test to make sure that it works okay. Also, I've done multiple hands-on testing of this feature.

### CODE REVIEW

**Comment by jtratner** [comment]
Seems reasonable - is there a reason to special case single directory?

Also can you add some examples to test_integration that makes sure it runs the way we expect?  (especially in terms of getting file name correct if we're not uploading to a container and instead to a subdirectory).

Something like:

```
stor.copytree([a, b], 'swift://A/C/some/subpath')
stor.copytree([a, c], 'swift://A/C/some/subpath2')
stor.copytree([a, b, c], 'swift://A/C/some/subpath')

```
where a and b are directories and c is a single file and also maybe determine what we should do if manifest already exists (I'm assuming overwrite?)

**Comment by pkaleta** [comment]
>Seems reasonable - is there a reason to special case single directory?

Yes. The reason to do that is backwards compatibility. The current behavior (`master` branch) is to create the manifest file inside the single directory that was passed. I want to preserve this behavior. So if a user does this:

```
stor.Path('swift://AUTH_foo/bar').upload([`logs`], use_manifest=True)
```

The manifest file will be created at `swift://AUTH_foo/bar/logs`.

In case of uploading multiple files, it's not immediately obvious where to create it. Imagine a situation where a user does this:

```
stor.Path('swift://AUTH_foo/bar').upload(['x.txt', '/tmp', 'logs/2016-02-02.log`], use_manifest=True)
```

There's a couple of ways around this:

- Create manifest in the current directory. This is the solution implemented by this branch. In this case the manifest will be created at `swift://AUTH_foo/bar/.data_manifest.csv`.
- Allow user to specify directory where the manifest is created (e.g. by using `use_manifest=<path to manifest file>`)

I'm going to add a comment that clarifies the current approach.

**Comment by pkaleta** [comment]
>Also can you add some examples to test_integration that makes sure it runs the way we expect? 

Good call. I've added integration test case in 8ad476ebef1933b1434361261355bd1a8ad1fc25.

**Comment by pkaleta** [comment]
friendly ping @jtratner

File: stor/swift.py [hunk 1]
```diff
@@ -1155,8 +1169,15 @@ def upload(self,
         # Convert everything to swift upload objects and prepend the relative
         # resource directory to uploaded results. Ignore the manifest file in the case of
         # since it will be uploaded individually
-        manifest_file_name = (Path(to_upload[0]) / utils.DATA_MANIFEST_FILE_NAME
-                              if use_manifest else None)
+        if use_manifest:
+            if len(to_upload) == 1 and os.path.isdir(to_upload[0]):
+                manifest_path_prefix = Path(to_upload[0])
+            else:
+                manifest_path_prefix = Path('.')
```
Comment (jtratner) [inline stor/swift.py]: and this is because we change directory before doing the uploading?

**Comment by pkaleta** [comment]
>Aside from the location of the manifest, these two commands would/do end up with the same files uploaded, right?

Correct.

>just in the second data manifest goes to swift://AUTH_foo/bar/.data_manifest.csv)?

Yes, however you'd have to pass `use_manifest=True` to `.upload()` for the manifest to be created.

File: stor/swift.py (hunk 1 above)
Comment (pkaleta) [inline stor/swift.py]: Correct.

**Comment by pkaleta** [comment]
Thanks @jtratner

**Comment by pkaleta** [comment]
@jtratner will merge, because I don't have permissions.

**Pull request merged**